components/findMatch.tsx

```diff
@@ -0,0 +1,112 @@
+import React, {Component} from "react";
+import { Button, TextField, ToggleButton, ToggleButtonGroup } from "@mui/material";
+
+export class FindMatch extends Component {
+
+    selected : boolean = false;
+    id : string = "";
+
+    changeStatus = () => {
+        this.selected = !this.selected;
+        this.updateView();
+    }
+
+    updateView = () => {
+        const el = document.getElementById("search");
+        
+        if(el){
+            el.innerHTML  = this.selected ? "Cancel" : "Find match";
+        }
+    }
+
+    makeRequest = (createGame : boolean) => {
+
+        //Send the first request to be added to the waiting list
+        const reqWaitingListOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ 
+                name : document.getElementById("playerName").value ,
+                waitingList : true, //Riguada una richista che ha a che fare con la waiting list 
+                accessWaitingList : createGame, // stabilisce se creare aggiungere o rimuovere dalla waiting list
+                id : this.id
+            })
+        };
+        
+        const url = "/api/game/findGame";
+        
+        fetch(url, reqWaitingListOptions)
+        .then(response => response.json())
+        .then(data => {
+            
+            if(createGame){
+                this.id = data.id;
+                //Send the second request to the server to create a game
+                const reqCreateGameOptions = {
+                    method: 'POST',
+                    headers: { 'Content-Type': 'application/json' },
+                    body: JSON.stringify({
+                        id : this.id,
+                        name : document.getElementById("playerName").value ,
+                        waitingList : false, //Riguada una richista che ha a che fare con la waiting list
+                        accessWaitingList : createGame // stabilisce se creare aggiungere o rimuovere dalla waiting list
+                    })
+                };
+                
+                //TODO : finire di completare la risposta della seconda richista (collegamento per generazione partita)
+                fetch(url, reqCreateGameOptions).then(response => response.json())
```

Comment:
i side effect (ossia questi fetch che vanno fare richieste), dovrebbero essere messi nei [saga](https://medium.com/nerd-for-tech/saga-pattern-971d59be3722), ed essere triggerati da delle azioni. 

praticamente hai delle azioni, e vai a triggerare le azioni che vanno a chiamare i saga che ti chiamano i fetch. Il risultato è messo nello store, che utilizzi per aggiornare quello che hai sullo schermo. 

Questo è il pattern che ho visto usare, non so se è la migliore esistente, ma fatto sta che rende molto chiaro la struttura, e separano la logica di business con quella di presentazione, sulla pagina noi vorremmo solamente la logica di presentazione. 

Anche se però lo devo ancora finire di fare...